### PR TITLE
Skip handling of container image lambda functions

### PR DESCRIPTION
**Pull request by duncanpharvey**
### What does this PR do?

Skip handling of container image Lambda functions. Lambda functions packaged into container images are instrumented with Datadog differently than Lambda functions packaged into a zip. The CloudFormation Macro should not make any changes to these container image Lambda functions.

### Motivation

https://github.com/DataDog/datadog-cloudformation-macro/issues/108

### Testing Guidelines

Created a SAM template with a zip Lambda function and container image Lambda function. Verified that (1) the zip Lambda function is instrumented with the CloudFormation Macro and sends metrics, logs, and traces to Datadog and (2) the container image Lambda function is unaffected by the CloudFormation Macro and sends metrics, logs, and traces to Datadog.

template.yaml

```
AWSTemplateFormatVersion: "2010-09-09"
Transform:
  - AWS::Serverless-2016-10-31
  - Name: DatadogServerlessTest
    Parameters:
      stackName: !Ref "AWS::StackName"
      apiKey: <redacted>
      pythonLayerVersion: "87"
      extensionLayerVersion: "52"

Globals:
  Function:
    Timeout: 10
    MemorySize: 256
    Architectures:
      - x86_64

Resources:
  HelloWorldFunctionContainer:
    Type: AWS::Serverless::Function
    Properties:
      PackageType: Image
      Environment:
        Variables:
          DD_LAMBDA_HANDLER: app.lambda_handler
          DD_API_KEY: <redacted>
          DD_SITE: datadoghq.com
          DD_TRACE_ENABLED: true
    Metadata:
      Dockerfile: Dockerfile
      DockerContext: ./hello_world
      DockerTag: python3.9-v1
  HelloWorldFunctionZip:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: hello_world/
      Handler: app.lambda_handler
      Runtime: python3.9
```

Dockerfile

```
FROM public.ecr.aws/lambda/python:3.9

COPY requirements.txt ./
RUN python3.9 -m pip install -r requirements.txt -t .

COPY app.py ./

COPY --from=public.ecr.aws/datadog/lambda-extension:latest /opt/extensions/ /opt/extensions

CMD ["datadog_lambda.handler.handler"]
```

### Types of changes

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [x] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: serverless/src/index.ts [hunk 1]
```diff
@@ -55,6 +55,7 @@ export interface FunctionProperties {
   TracingConfig?: { [key: string]: string };
   FunctionName?: string;
   Architectures?: [string];
+  PackageType?: string;
```
Comment (duncanista) [inline serverless/src/index.ts:58]: Is this property requested by AWS for Container based Lambdas?

Comment (duncanista) [reply serverless/src/index.ts:58]: Just looked at the docs https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html, so it looks like yes.

**Review by duncanista — APPROVED** [review]
Thanks for fixing this! LGTM

File: serverless/src/index.ts (hunk 1 above)
Comment (duncanpharvey) [reply serverless/src/index.ts:58]: Correct, it's set to `Image` for Container based Lambdas.

**Pull request merged**